### ISSUE TITLE
Add support for per-step update frequency

### ISSUE THREAD

**Issue by Crocmagnon** [topgrade-rs/topgrade#375]
## I want to suggest some general feature
I run topgrade frequently on my machine, to keep up to date with most of my packages. There are a few however that I don't mind being a bit behind. For example `tlmgr` (TeX). I don't need to update my TeX packages multiple times a week, but once in a while I'd like to.

My proposal would be to allow a config similar to this:

```toml
# topgrade.toml
[update_frequency]
daily = ["rustup"]
weekly = ["fisher", "krew"]
monthly = ["tlmgr"]
```

We would probably need to record the last run for each step. Then:
* On a new run, check if a step belongs to one known update frequency.
    * If not, run the step.
    * If yes, check if enough time has elapsed.
        * If yes, then run the step.
        * If not, skip it and mention it to the user in the summary.

E.g., assuming current date is 2023-03-03 10:13

```
── 12:03:11 - Summary ──────────────────────────────────────────────────────────
Brew (ARM): OK
Brew Cask (ARM): OK
fisher: Skipped (last run: 2022-03-01 15:45, update weekly)
TLDR: OK
rustup: OK
cargo: OK
pipx: OK
tlmgr: Skipped (last run: 2022-02-13 09:32, update monthly)
chezmoi: OK
npm: OK
Containers: OK
krew: Skipped (last run: 2022-03-01 15:45, update weekly)
GitHub CLI Extensions: OK
System upgrade: OK
```

Packages not mentioned in any update frequency would default to "every time".

**Comment by tomadojuice** [comment]
I'll have a look at this and will try to implement it. Might take some time though ...

**Comment by SteveLauC** [comment]
> I'll have a look at this and will try to implement it. Might take some time though ...

Hi @tomadojuice, Thanks for being interested in implementing this feature:)

First, I would like to say it is kinda hard (at least for now) to implement this feature for the following reasons:
1. Skipped step won't be shown in the summary unless `--verbose` option is present
2. The error and step finish state handling in the current `topgrade` implementation is messy, and you can find it is very inconsistent, it varies from step to step (I do have a refactoring plan in the future, but it will take some time I guess)

But if you still want to implement it, here are some tips may get you started:
1. You need to define a new configuration section in `src/config.rs`
    ```rust
    /// Configuration file
    pub struct ConfigFile {
        // existing sections

        // new section
        update_frequency: Option<UpdateFrequency>
    }
    struct UpdateFrequency {
        daily: Optoin<Vec<Step>>,
        weekly: Option<Vec<Step>>,
        monthly:  Option<Vec<Step>>,
    }
    ```
   
    Note that all these field are optional, means they can be absent from the configuration file, this is some kind of convention that `topgrade` uses.

2. You may need to store the timestamp of a step being updated lasted updated in some file (and in a format) to make it persistent
    
    It should be stored under the [data_dir](https://docs.rs/etcetera/latest/etcetera/app_strategy/trait.AppStrategy.html#tymethod.data_dir)
     
3. For this logic
    
    > On a new run, check if a step belongs to one known update frequency.
If not, run the step.
If yes, check if enough time has elapsed.
If yes, then run the step.
If not, skip it and mention it to the user in the summary.

    It should be implemented inside of [Runner::execute()](https://github.com/topgrade-rs/topgrade/blob/a0ff565220e3e79fd67c6e20c30edeff4d62d19f/src/runner.rs#L25) I guess

Hope this could help you:)
